### PR TITLE
Replacing Diesel by Sqlx

### PR DESCRIPTION
**Pull request by fulmicoton**
- replaces diesel by sqlx
- removes deps (chrono, openssl)
- removes postgresql redundant testsuite (not sure why it what there)
- improved logging in postgres metastore.
- updated documentation.

Closes https://github.com/quickwit-oss/quickwit/issues/1587

### CODE REVIEW

**Comment by fulmicoton** [comment]
before ldd `which quickwit`
```
	linux-vdso.so.1 (0x00007ffdff1e5000)
	libpq.so.5 => /lib/x86_64-linux-gnu/libpq.so.5 (0x00007fa38f683000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fa38b3d4000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fa38b3b4000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fa38f67e000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fa38f679000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fa38b2cd000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fa38f672000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fa38b0a5000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fa38f6eb000)
	libssl.so.3 => /lib/x86_64-linux-gnu/libssl.so.3 (0x00007fa38b001000)
	libcrypto.so.3 => /lib/x86_64-linux-gnu/libcrypto.so.3 (0x00007fa38abbf000)
	libgssapi_krb5.so.2 => /lib/x86_64-linux-gnu/libgssapi_krb5.so.2 (0x00007fa38ab6b000)
	libldap-2.5.so.0 => /lib/x86_64-linux-gnu/libldap-2.5.so.0 (0x00007fa38ab0c000)
	libkrb5.so.3 => /lib/x86_64-linux-gnu/libkrb5.so.3 (0x00007fa38aa41000)
	libk5crypto.so.3 => /lib/x86_64-linux-gnu/libk5crypto.so.3 (0x00007fa38aa12000)
	libcom_err.so.2 => /lib/x86_64-linux-gnu/libcom_err.so.2 (0x00007fa38aa0c000)
	libkrb5support.so.0 => /lib/x86_64-linux-gnu/libkrb5support.so.0 (0x00007fa38a9fe000)
	liblber-2.5.so.0 => /lib/x86_64-linux-gnu/liblber-2.5.so.0 (0x00007fa38a9eb000)
	libsasl2.so.2 => /lib/x86_64-linux-gnu/libsasl2.so.2 (0x00007fa38a9d0000)
	libgnutls.so.30 => /lib/x86_64-linux-gnu/libgnutls.so.30 (0x00007fa38a7e5000)
	libkeyutils.so.1 => /lib/x86_64-linux-gnu/libkeyutils.so.1 (0x00007fa38a7de000)
	libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007fa38a7ca000)
	libp11-kit.so.0 => /lib/x86_64-linux-gnu/libp11-kit.so.0 (0x00007fa38a68d000)
	libidn2.so.0 => /lib/x86_64-linux-gnu/libidn2.so.0 (0x00007fa38a66c000)
	libunistring.so.2 => /lib/x86_64-linux-gnu/libunistring.so.2 (0x00007fa38a4c2000)
	libtasn1.so.6 => /lib/x86_64-linux-gnu/libtasn1.so.6 (0x00007fa38a4aa000)
	libnettle.so.8 => /lib/x86_64-linux-gnu/libnettle.so.8 (0x00007fa38a464000)
	libhogweed.so.6 => /lib/x86_64-linux-gnu/libhogweed.so.6 (0x00007fa38a41c000)
	libgmp.so.10 => /lib/x86_64-linux-gnu/libgmp.so.10 (0x00007fa38a398000)
	libffi.so.8 => /lib/x86_64-linux-gnu/libffi.so.8 (0x00007fa38a38b000)

```

after ldd target/release/quickwit
```
	linux-vdso.so.1 (0x00007ffe285fb000)
	libssl.so.3 => /lib/x86_64-linux-gnu/libssl.so.3 (0x00007f6213357000)
	libcrypto.so.3 => /lib/x86_64-linux-gnu/libcrypto.so.3 (0x00007f6212f15000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f6212ce9000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f6212cc9000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f6212be2000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f62129b8000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f6215e8a000)
```

File: quickwit-metastore/src/tests.rs [hunk 1]
```diff
@@ -1997,119 +1991,54 @@ macro_rules! metastore_test_suite {
...
             async fn test_metastore_delete_source() {
+                let _ = tracing_subscriber::fmt::try_init();
                 crate::tests::test_suite::test_metastore_delete_source::<$metastore_type>().await;
             }
         }
-    };
-}
-
-#[cfg(feature = "postgres")]
-macro_rules! metastore_test_suite_for_postgresql {
```
Comment (fulmicoton) [inline quickwit-metastore/src/tests.rs:2046]: I have no clue why this thing exists.
As a result 2 unit tests were not running on postgresql. (add_source and delete_source).

File: quickwit-metastore/migrations/postgresql/1_create-indexes.up.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
+CREATE TABLE indexes (
```
Comment (evanxg852000) [inline quickwit-metastore/migrations/postgresql/1_create-indexes.up.sql:1]: My guess is that we are ok this migration will not upgrade a previously running version of Quickwit. Users might need to manually upgrade or create a new db and then export/import the metastore data.

Comment (evanxg852000) [inline quickwit-metastore/migrations/postgresql/1_create-indexes.up.sql:1]: My guess is that we are ok this migration will not upgrade a previously running version of Quickwit. Users might need to manually upgrade or create a new db and then export/import the metastore data.

File: quickwit-metastore/src/metastore/postgresql_metastore.rs [hunk 3]
```diff
@@ -23,622 +23,556 @@ use std::sync::Arc;
...
+        .iter()
+        .map(|split_id| split_id.as_str())
+        .collect();
+    let unaffected_ids_set: HashSet<&str> = split_ids
+        .iter()
+        .copied()
+        .filter(|&split_id| !affected_ids_set.contains(split_id))
+        .collect();
+
+    // SQL query query that helps figure out if index exist, non-existent
```
Comment (evanxg852000) [inline quickwit-metastore/src/metastore/postgresql_metastore.rs:269]:
```suggestion
    // SQL query that helps figure out if index exist, non-existent
```
```suggestion
    // SQL query query that helps figure out if index exist, non-existent
```

Comment (evanxg852000) [inline quickwit-metastore/src/metastore/postgresql_metastore.rs:269]:
```suggestion
    // SQL query that helps figure out if index exist, non-existent
```
```suggestion
    // SQL query query that helps figure out if index exist, non-existent
```

**Review by evanxg852000 — APPROVED** [review]
LGTM :+1:

File: docs/get-started/installation.md [hunk 4]
```diff
@@ -33,23 +32,23 @@ You can install these dependencies using the following command:
 <TabItem value="ubuntu" label="Ubuntu">
 
 ```bash
-apt-get -y update && apt-get -y install libpq-dev libssl-dev
+apt-get -y update && apt-get -y install libssl
```
Comment (fulmicoton) [inline docs/get-started/installation.md:35]:
```suggestion
apt-get -y update && apt-get -y install libssl3
```

File: docs/get-started/installation.md [hunk 5]
```diff
@@ -33,23 +32,23 @@ You can install these dependencies using the following command:
...
+apt-get -y update && apt-get -y install libssl3
 ```
 
 </TabItem>
 
 <TabItem value="aws-linux" label="AWS Linux">
 
 ```bash
-yum -y update && yum -y install postgresql-libs openssl-devel
+yum -y update && yum -y install openssl
```
Comment (fulmicoton) [inline docs/get-started/installation.md:43]: Confirm that openssl is the right name of the package?

**Pull request merged**